vulcan/lib/client/jsx/components/workflow/workflow_manager.tsx

```diff
@@ -101,9 +103,14 @@ export default function WorkflowManager({
     let session = {
       ...defaultSession,
       workflow_name: cwlName(workflowName) || workflowName,
-      project_name: projectName
+      project_name: projectName,
+      inputs: workflow ? defaultInputs(workflow) : defaultSession.inputs
     };
+    // To allows correct values to pass through, set session before letting auto-pass trigger.
     dispatch(setSession(session));
+    if (workflow && workflow.vignette?.includes("Primary inputs are skippable") && completedSteps(workflow, state.status).length<1) {
```

Comment:
Hm.... 

I think the first one should be handling new figures that haven't been saved, but that you've edited and run in your browser. In which case, `figureId` would be `null`. For any saved figure, it should always have a `figureId`.

I think the second case is for any old figure that would not have a snapshot taken. That should theoretically not happen in production, since I ran a backfill command, so that all figures have snapshots in the database. But I guess it would also help with development (if you haven't run that command), or if you have a production session in your browser that hasn't been updated to the newer schema...

Reply:
Ahh gotcha!

Case1, figures that haven't been saved... in which case the "standard" at the moment is then needing to click `Run` to actually get all the way back.  I'll leave that, but we could hi-jack the auto-pass system here easily if we anted to adjust!

Case2, the completedSteps check I added in the last commit was aimed at this case, but in such cases, I think the ~`inputs: defaultInputs(workflow)` code changed just above (workflow_manager.tsx L107) would both 1) erase any changes primary inputs, and 2) remove all later stepUI choices!  I can probably add logic to skip setting `input` in such cases.  I wonder if there's a good way to test once I implement the check...

Reply:
> I can probably add logic to skip setting input in such cases.

Yeah, I think that would make sense, since it might throw users off if their inputs are unexpectedly reset to the defaults ... they would basically lose work, if that happened?

You could probably test locally if you have some version of the figure prior to the snapshot system, or manually delete the snapshot in the database. Would be nice to have automated Jest / UI tests too, where it should be possible to edit the test fixtures to mimic that kind of state...

Reply:
Hmmmm on second thought seems like:
- case1 where there is no figure_id, so we're talking about the new session / create figure path, if the person clicks `cancel` on the confirm box, they would expect a clean start with only the base workflow defaults.  Thus no locally stored inputs pass through is needed.
- case2, where there is no workflow snapshot, the `selectWorkflowSnapshot()` call automatically creates a snapshot from state + the workflow definition, so long as `workflowByName()` can find the workflow_name->workflow definition.  So lines 116 & 117 are only hit if the requested workflow_name doesn't exist which.... would cause other issues to say the least!  Thus there's still no locally stored inputs pass through needed!

Gonna revert my last commit which added it, then I think this'll be ready to merge!